tasks/realVVNCServerConfigUpdater.go

```diff
@@ -0,0 +1,339 @@
+//go:build !windows
+// +build !windows
+
+package tasks
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/sirupsen/logrus"
+
+	tacoexec "github.com/cloudradar-monitoring/tacoscript/exec"
+	"github.com/cloudradar-monitoring/tacoscript/realvnc"
+	"github.com/cloudradar-monitoring/tacoscript/utils"
+)
+
+const (
+	DefaultMacExecPath           = `/Library/vnc`
+	DefaultMacExecName           = `vncserver`
+	DefaultLinuxExecPath         = `/usr/bin`
+	DefaultLinuxExecCmd          = `vncserver-x11`
+	DefaultLinuxLicenseReloadCmd = `vnclicense`
+
+	DefaultConfigFilePermissions = 0644
+)
+
+func (rvste *RealVNCServerTaskExecutor) applyConfigChanges(rvst *RealVNCServerTask) (addedCount int, updatedCount int, err error) {
+	configValues, tempFile, err := newConfigValuesWithTempOutputFile(rvst)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// make sure to close the writer
+	defer func() {
+		closeErr := tempFile.Close()
+		// if we're already returning an err then that's more important than the closeErr
+		if err == nil {
+			// if no existing err and a closeErr then return the closeErr
+			err = closeErr
+		}
+	}()
+
+	addedCount, updatedCount, err = rvste.makeChanges(rvst, configValues)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	err = commitChanges(rvst, tempFile)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	return addedCount, updatedCount, nil
+}
+
+func newConfigValuesWithTempOutputFile(rvst *RealVNCServerTask) (
+	configValuesFile *realvnc.ConfigValues, tempFile *os.File, err error) {
+	configFilename := rvst.ConfigFile
+	if configFilename == "" {
+		return nil, nil, errors.New(ErrConfigFileMustBeSpecifiedMsg)
+	}
+
+	logrus.Debugf("reading config values from %s", configFilename)
+	configValuesFile, err = realvnc.NewConfigValuesFromFile(configFilename)
+	if err != nil {
+		// we can continue ok if no existing config file
+		if !errors.Is(err, os.ErrNotExist) {
+			return nil, nil, err
+		}
+	}
+
+	baseConfigFilename := filepath.Base(rvst.ConfigFile)
+	tempFile, err = os.CreateTemp("", baseConfigFilename)
```

Comment:
Not sure why a temp file is needed here. Couldn't you just write the config to `bytes.Buffer` and dump that into the final file?

Reply:
agreed. changed to use a `bytes.Buffer`. done.